torchtext/models/roberta/bundler.py

```diff
@@ -97,6 +105,45 @@ def get_model(self, head: Optional[Module] = None, load_weights: bool = True, fr
             model.load_state_dict(state_dict, strict=True)
         return model
 
+    @classmethod
+    def from_config(
+        self,
+        config: RobertaEncoderConf,
+        head: Optional[Module] = None,
+        freeze_encoder: bool = False,
+        checkpoint: Optional[Union[str, Dict[str, torch.Tensor]]] = None,
+        *,
+        dl_kwargs: Dict[str, Any] = None,
+    ) -> RobertaModel:
+        """Class method to intantiate model with user-defined encoder configuration and checkpoint
+
+        Args:
+            config (RobertaEncoderConf): An instance of class RobertaEncoderConf that defined the encoder configuration
+            head (nn.Module, optional): A module to be attached to the encoder to perform specific task
+            freeze_encoder (bool): Indicates whether to freeze the encoder weights
+            checkpoint (str or Dict[str, torch.Tensor], optional): Path to or actual model state_dict. state_dict can have partial weights i.e only for encoder.
+            dl_kwargs (dictionary of keyword arguments): Passed to :func:`torch.hub.load_state_dict_from_url`.
+        """
+        model = _get_model(config, head, freeze_encoder)
+        if checkpoint is not None:
+            if torch.jit.isinstance(checkpoint, Dict[str, torch.Tensor]):
+                state_dict = checkpoint
+            elif isinstance(checkpoint, str):
+                dl_kwargs = {} if dl_kwargs is None else dl_kwargs
+                state_dict = load_state_dict_from_url(checkpoint, **dl_kwargs)
+            else:
+                raise TypeError("checkpoint must be of type `str` or `Dict[str, torch.Tensor]` but got {}".format(type(checkpoint)))
+            if head is not None:
+                regex = re.compile(r"^head\.")
+                head_state_dict = {k: v for k, v in model.state_dict().items() if regex.findall(k)}
+                # if not all the head keys are present in checkpoint then we shall update the state_dict with the provided head state_dict
+                if not all(key in state_dict.keys() for key in head_state_dict.keys()):
+                    state_dict.update(head_state_dict)
```

Comment:
@mthrok I have updated the logic how to handle the case when checkpoint contains the head parameters or not.

Case1: Does not contains head parameters: In this case, we would update the state with supplied head state dict. This would mean that we are keeping intact the head module whatever is provided by the user. This is inline with what we discussed here https://github.com/pytorch/text/pull/1424#discussion_r737018029

Case2: Does contain head parameters: In this case we would use the checkpoint as such.

Note that in both the cases, now we have made `strict=True` to ensure all key matches. Let me know if this sounds reasonable to contain various edge cases?